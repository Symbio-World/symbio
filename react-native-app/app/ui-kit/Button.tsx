import * as React from 'react'
import { TouchableOpacity, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'

import { Icon, IconsEnum } from './Icon'

export enum ButtonStyle {
  FILL,
  OUTLINE,
  TRANSPARENT,
}

type Props = {
  onPress?: () => void
  title?: string
  icon?: IconsEnum
  variant?: ButtonStyle
}

const ButtonBaseStyle = {
  [ButtonStyle.FILL]: [t.bgPrimary],
  [ButtonStyle.OUTLINE]: [t.bgWhite, t.border2, t.borderPrimary],
  [ButtonStyle.TRANSPARENT]: [t.bgTransparent],
}

const InnerTitleBaseStyle = {
  [ButtonStyle.FILL]: [t.textWhite],
  [ButtonStyle.OUTLINE]: [t.textPrimary],
  [ButtonStyle.TRANSPARENT]: [t.textPrimary],
}

export const Button: React.FC<Props> = ({
  onPress = () => {},
  title = '',
  variant = ButtonStyle.FILL,
  icon = null,
}) => {
  const buttonStyle = ButtonBaseStyle[variant]
  const innerStyle = [InnerTitleBaseStyle[variant]]
  return (
    <TouchableOpacity
      style={[
        t.p4,
        t.itemsCenter,
        t.justifyCenter,
        t.flex,
        t.flexRow,
        t.rounded,
        ...buttonStyle,
      ]}
      onPress={onPress}
    >
      {icon ? <Icon style={[t.mR2]} iconType={icon} /> : null}
      <Text style={[t.textCenter, t.text2xl, t.textWhite, ...innerStyle]}>
        {title}
      </Text>
    </TouchableOpacity>
  )
}
