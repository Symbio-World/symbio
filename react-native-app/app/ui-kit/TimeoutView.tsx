import * as React from 'react'
import { View, Text } from 'react-native'
import { t } from 'react-native-tailwindcss'
import { Button } from './Button'

type Props = {
  onRetry: () => void
}
export const TimeoutView: React.FC<Props> = ({ onRetry }) => {
  return (
    <View style={[t.flex1, t.justifyCenter, t.alignCenter, t.p8]}>
      <Text style={[t.mB2]}>
        It took to long to receive information, please check your network
        connection and retry
      </Text>
      <Button title="Retry" onPress={onRetry} />
    </View>
  )
}
