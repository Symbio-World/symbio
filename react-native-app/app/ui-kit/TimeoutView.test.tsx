import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { TimeoutView } from './TimeoutView'
import { Button } from './Button'

describe('TimeoutView', () => {
  it('renders correctly', () => {
    const handleRetry = jest.fn()
    const { toJSON } = render(<TimeoutView onRetry={handleRetry} />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('triggers onRetry prop', () => {
    const handleRetry = jest.fn()
    const { getByType } = render(<TimeoutView onRetry={handleRetry} />)
    fireEvent(getByType(Button), 'onPress')
    expect(handleRetry).toBeCalledTimes(1)
  })
})
