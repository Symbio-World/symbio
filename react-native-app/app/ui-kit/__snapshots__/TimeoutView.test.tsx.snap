// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeoutView renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "textAlignVertical": "center",
      },
      Object {
        "padding": 32,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "marginBottom": 8,
        },
      ]
    }
  >
    It took to long to receive information, please check your network connection and retry
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B35340",
        "borderRadius": 4,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 24,
          },
          Object {
            "color": "#fff",
          },
          Array [
            Object {
              "color": "#fff",
            },
          ],
        ]
      }
    >
      Retry
    </Text>
  </View>
</View>
`;
